pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -387,6 +401,9 @@ func (v *VolumeOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, out,
 		}
 		v.AddOpts.ClaimSize = q.String()
 	}
+	if len(v.AddOpts.DefaultMode) == 0 {
+		v.AddOpts.DefaultMode = "644"
```

Comment:
I'm ok with either solution, but just a comment: default flag value and "user provided the flag" are two different things. 

To know when the user used a flag, you use the `Changed` attribute in the flag, or `Changed(flagName)` in the flag set. It will correctly reflect the case even if the user provided the flag with the same value set as default.